ddtrace/profiling/_threading.pyx

```diff
@@ -10,9 +10,20 @@ from six.moves import _thread
 from ddtrace import _threading as ddtrace_threading
 
 
-IF UNAME_SYSNAME == "Linux":
-    from cpython cimport PyLong_FromLong
+from cpython cimport PyLong_FromLong
+
+
+cdef extern from "<Python.h>":
+    # This one is provided as an opaque struct from Cython's
+    # cpython/pystate.pxd, but we need to access some of its fields so we
+    # redefine it here.
+    ctypedef struct PyThreadState:
```

Comment:
I'm confused about this.  `ctypedef` is defining the C layout for a struct named PyThreadState as a struct with a single `unsigned long` field.

On the other hand, cpython's internal definition for PyThreadState is a typedef to the `_ts` [struct in cpython](https://github.com/python/cpython/blob/main/Include/cpython/pystate.h#L64).

If one were to use this struct to bring a PyThreadState object into cpython, then on Linux this redefinition will pun the `PyThreadState *next` to a 64-bit integer (and on Windows, it'll take the bottom 32 bits of the pointer because a long is 32-bits there) called `thread_id`.  This will be a moderately unique value, but it isn't the same as PyThreadState's `thread_id`.